src/compiler/moduleNameResolver.ts

```diff
@@ -1202,8 +1206,13 @@ namespace ts {
             if (traceEnabled) {
                 trace(host, Diagnostics.Resolution_for_module_0_was_found_in_cache_from_location_1, moduleName, containingDirectory);
             }
-            return { value: result.resolvedModule && { path: result.resolvedModule.resolvedFileName, extension: result.resolvedModule.extension, packageId: result.resolvedModule.packageId } };
+            return {

```

Comment:
Can we instead pass failed lookup locations here so that we do not create another objects just to unwrap this one and add failed lookup locations at caller site?

Reply:
we also get to keep failed lookup locations as Push<string> then right?